ni_measurementlink_service/_internal/parameter/serialization_strategy.py

```diff
@@ -107,12 +156,80 @@ def vector_decoder(field_index, key):
     return vector_decoder
 
 
+def _decode_varint(buffer, pos):
+    mask = (1 << 64) - 1
+    result_type = int
+    result = 0
+    shift = 0
+    while 1:
+        b = buffer[pos]
+        result |= (b & 0x7F) << shift
+        pos += 1
+        if not (b & 0x80):
+            result &= mask
+            result = result_type(result)
+            return (result, pos)
+        shift += 7
+        if shift >= 64:
+            raise Exception("Too many bytes when decoding varint.")
+
+
+def _inner_message_decoder(field_index, is_repeated, is_packed, key, new_default):
+    """Based on google.protobuf.internal.MessageDecoder."""
+
+    def _convert_to_byte_string(memview):
+        """Convert bytes to byte_string."""
+        byte_str = memview.tobytes()
+        return byte_str
+
+    def decode_message(buffer, pos, end, message, field_dict):
+        value = field_dict.get(key)
+        if value is None:
+            value = field_dict.setdefault(key, new_default(message))
+        # Read length.
+        (size, pos) = _decode_varint(buffer, pos)
+        new_pos = pos + size
+        thestring = _convert_to_byte_string(buffer[pos:new_pos])
+        value.ParseFromString(thestring)
+        return new_pos
+
+    return decode_message
+
+
+def _double_xy_data_decoder(decoder) -> Callable[[int, str], Callable]:
+    """Decoder for DoubleXYData.
+
+    Args
+    ----
+        decoder (Callable[[int, bool, bool], Callable]): Specific decoder(Callable) that takes in
```

Comment:
Don't duplicate type hints in the docstring. Sphinx can parse type hints.

You don't really need to write docstrings for private/internal stuff. Sphinx doesn't generate any docs for these functions. If you do write docstrings for private/internal stuff, please try to avoid burying the content in boilerplate.